machine/v1/types_controlplanemachineset.go

```diff
@@ -284,6 +289,40 @@ type GCPFailureDomain struct {
 	Zone string `json:"zone"`
 }
 
+// OpenStackFailureDomain configures failure domain information for the OpenStack platform.
+// +kubebuilder:validation:MinProperties:=1
+type OpenStackFailureDomain struct {
+	// availabilityZone is the nova availability zone in which the OpenStack machine provider will create the VM.
+	// If not specified, the VM will be created in the default availability zone specified in the nova configuration.
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// rootVolume contains settings that will be used by the OpenStack machine provider to create the root volume attached to the VM.
+	// If not specified, no root volume will be created.
+	// If specified without a Zone, the root volume will be created in the availability zone specified by the volume type. If the volume type
+	// (configured in the OpenStack cluster) does not specify an availability zone, the root volume will be created in the default availability
+	// zone specified in the cinder configuration. See https://docs.openstack.org/cinder/latest/admin/availability-zone-type.html for more details.
+	// If the OpenStack cluster is deployed with the cross_az_attach configuration option set to false, the root volume will have to be in the same
+	// availability zone as the VM (defined by availabilityZone).
+	// +optional
+	RootVolume *RootVolume `json:"rootVolume,omitempty"`
+}
+
+// RootVolume represents the volume metadata to boot from.
+// The original RootVolume struct is defined in the v1alpha1 but it's not best practice to use it directly here so we define a new one
+// that should stay in sync with the original one.
+// +kubebuilder:validation:MinProperties:=1
+type RootVolume struct {
+	// availabilityZone specifies the Cinder availability where the root volume will be created.
+	// If not specifified, the root volume will be created in the availability zone specified by the volume type. If the volume type
```

Comment:
To reviewers: I put this comments here before in the very near future @pierreprinetti will add the `Type` field too, which will make the Zone field not required.

Reply:
What about `If not specifified, the root volume will be created in the availability zone specified by the volume type.` -> `If not specifified, the root volume will be created in the availability zone specified by the volume type in the cinder configuration.`? To stress the volume type configuration is currently available only through cinder? To avoid anyone reading the description from searching for the volume type field.

Reply:
ok, I'll change the text.